atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/table/common/TableTasks.java

```diff
@@ -480,4 +479,40 @@ PartialDiffStats call(RangeRequest request, MutableRange range, DiffStrategy str
         private long rowsCopied = 0;
         private long cellsCopied = 0;
     }
+
+    private static class InterruptibleRangeExecutor {
+        private final ExecutorService exec;
+        private final int batchSize;
+        private final int threadCount;
+
+        InterruptibleRangeExecutor(ExecutorService exec, int batchSize, int threadCount) {
+            this.exec = exec;
+            this.batchSize = batchSize;
+            this.threadCount = threadCount;
+        }
+
+        void executeTask(String taskName, InterruptibleRangeTask task) throws InterruptedException {
+            BlockingWorkerPool pool = new BlockingWorkerPool(exec, threadCount);
+            for (final MutableRange range : getRanges(threadCount, batchSize)) {
+                if (Thread.currentThread().isInterrupted()) {
+                    log.info("Thread interrupted. Cancelling {} of range {}", taskName, range);
+                    break;
+                }
+                pool.submitTask(() -> {
+                    do {
+                        if (Thread.currentThread().isInterrupted()) {
+                            log.info("Thread interrupted. Cancelling {} of range {}", taskName, range);
+                            break;
```

Comment:
this should probably return as opposed to just `break` to prevent future changes below loop impacting interruption